### ISSUE TITLE
[quest] End of the road?

### ISSUE THREAD

**Issue by mpikzink** [streetcomplete/StreetComplete#3705]
### General
Affected tag(s) to be modified/added: 
Very last node node at the end of a highway=*

Question asked: **What is at this end of this road?**
 - "a turning loop with a non-traversable island" : [highway=turning_circle](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dturning_loop)
 - "nothing, dead end" : [noexit=yes](https://wiki.openstreetmap.org/wiki/Key:noexit)
 - "a turning circle" : [highway=turning_circle](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dturning_loop) 
 - "this is not the end of the road, it continues..." 
   -  So you see the continueing road on the map?
   - yes : bug=Nodes are close to each other but not connected
   -  No : [fixme=continue](https://wiki.openstreetmap.org/wiki/DE:Key:fixme)

 - "the road ends but there is a footpath..." : note=path ist missing
### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/streetcomplete/StreetComplete/wiki/Adding-new-Quests-to-StreetComplete)):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary
- [x] 💤 Not an overwhelming percentage of quests have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of map data (Worth the effort)

### Ideas for implementation

**Element selection:** 
Find the **begin** and **end** node of a way with the key `highway=*`. This nodes need to be connected to another way or should have `noexit=yes` or should have `fixme=continue` .

[See discussion](https://github.com/streetcomplete/StreetComplete/discussions/3702)

**Comment by HolgerJeromin** [comment]
We should skip service=driveways and parking_aisles

**Comment by andrewharvey** [comment]
should skip roads with end with a `highway=turning_circle`.

**Comment by andrewharvey** [comment]
Thinking about edge cases like when a pathway leads into a park, it probably would trigger this but neither tag would make sense. You'd want to exclude paths that end inside areas like `amenity=park`, `natural=beach`

**Comment by 10992-osm** [comment]
Another thing, particularly with paths and other less important highways is that the path might gradually fade out into an area where people walk as they like, particularly in parks or beaches, as mentioned by @andrewharvey.  However, parks, at least, are also likely (or as likely as anywhere else) to contain paths that need continuing.

It also shouldn't be asked for highways whose final node is on a `building` way.

Not sure about `barrier` nodes.  If it is a gate, it will continue in some way the vast majority of the time.  For farmers' fields, it might not really continue as any path/track worth mapping, though.  With some other barriers, it might stop there.  It may still be worth asking in these cases.

> should skip roads which end with a `highway=turning_circle`.

What if there is a footpath that continues that hasn't already been mapped?  We would want to find that.

**Comment by matkoniecz** [comment]
> What if there is a footpath that continues that hasn't already been mapped? We would want to find that.

There is much lower chance of that and such tagging is going to be less likely to be welcome.

Overall I am still unsure about this quest, but aggressive filtering any dubious cases is likely preferable.

**Comment by westnordost** [comment]
Hmm, it would be wrong to add `noexit=yes` where there is a turning circle in reality, right? Same with turning loops... and maybe other things(?).

So, the quest would need to ask something like "What is at this end of this road?" and the user can answer "nothing, dead end", "a turning loop", "a turning circle", "this is not the end of the road, it continues...", "the road ends but there is a footpath..." and/or more.

**Comment by mpikzink** [comment]
> Hmm, it would be wrong to add `noexit=yes` where there is a turning circle in reality, right? Same with turning loops... and maybe other things(?).
> 
> So, the quest would need to ask something like "What is at this end of this road?" and the user can answer "nothing, dead end", "a turning loop", "a turning circle", "this is not the end of the road, it continues...", "the road ends but there is a footpath..." and/or more.

OK, i didn't know thus turning nodes. I will rewrite fhe proposal!

**Comment by westnordost** [comment]
Still, I am kind of "😬" about this, I fear we might miss something. I think what should be done first is some research (with an overpassquery ) to find such situations and Stichprobenartig see what situations can in reality exist.

**Comment by HolgerJeromin** [comment]
I think we will find many unconnected footways ending before a house like
https://www.openstreetmap.org/way/220513514
![image](https://user-images.githubusercontent.com/2410353/152220166-bbe6a59b-85e7-44b6-82f5-597b54a241e8.png)

**Comment by mpikzink** [comment]
What is the argument against the noexit=yes tag here? The tag would at least explicitly define the end of the path.

**Comment by 10992-osm** [comment]
> Hmm, it would be wrong to add `noexit=yes` where there is a turning circle in reality, right? Same with turning loops... and maybe other things(?).

I don't think it would be wrong.  Ways can still continue after turning circles, either as a lesser highway, or even the same type.

I've looked up some turning circles.

1. https://www.openstreetmap.org/node/826166797: This is a turning circle at the end of a road, with no other paths mapped.  From aerial imagery, I suspect there may be a footpath (or two, it's more of a T than a circle) that connect(s) with the main(er) road.  If I was there, I would want to be able to tag either `noexit=yes` or `fixme=continue`.
1. https://www.openstreetmap.org/node/6847503010: Already has a path mapped, but if it hadn't it might be the same as the above.
1. https://www.openstreetmap.org/node/828943779: This node is definitely a turning circle with no exit, which we would want to tag `noexit=yes`.

> So, the quest would need to ask something like "What is at this end of this road?" and the user can answer "nothing, dead end", "a turning loop", "a turning circle", "this is not the end of the road, it continues...", "the road ends but there is a footpath..." and/or more.

---

> ### General
> 
> Affected tag(s) to be modified/added: Very last node node at the end of a highway=*
> 
> Question asked: **What is at this end of this road?**
> 
>     * "a turning loop with a non-traversable island" : [highway=turning_circle](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dturning_loop)
> 
>     * "nothing, dead end" : [noexit=yes](https://wiki.openstreetmap.org/wiki/Key:noexit)
> 
>     * "a turning circle" : [highway=turning_circle](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dturning_loop)
> 
>     * "this is not the end of the road, it continues..."
>       
>       * So you see the continueing road on the map?
>       * yes : bug=Nodes are close to each other but not connected

Would that be better as a `fixme`?  Close ways are often detected by QA tools anyway, but we'd need to have a tagging for that situation.

>     * "the road ends but there is a footpath..." : note=path ist missing

Wouldn't `fixme=path is missing` or `fixme=continues as path` or `fixme=continues as another highway type` be better?

**Comment by matkoniecz** [comment]
> I don't think it would be wrong.

I am not exactly worrying that it would strictly invalid. I rather worry that it would be pointless/unwanted/disliked/irritating/unwelcome.

**Comment by westnordost** [comment]
@HolgerJeromin , I think there is a misunderstanding. I think this quest is supposed to be asked for **endpoints of roads**, not of paths. Also, when a turning circle is present, the quest should **not** be asked.

**Comment by HolgerJeromin** [comment]
>  I think this quest is supposed to be asked for endpoints of roads, not of paths.

Fine with me, but we should adjust the description of the issue in that case:

>> Very last node node at the end of a highway=*

and

>> Element selection: Find the begin and end node of a way with the key `highway=*`.

**Comment by peternewman** [comment]
> Stichprobenartig

Random/randomly (which wasn't part of my schoolboy German!)

**Comment by fnordiff** [comment]
Perhaps two things have been mixed up here that should be treated separately?
* The [discussion](https://github.com/streetcomplete/StreetComplete/discussions/3702) is about "paths in the woods". That means "non-urban" (while urban is the area where the current proposal would trigger a lot of useless tasks and/or floods of unnecessary noexit=yes)
* In the above suggestion, the default answers provided include "turning_loop" and "turning_circle" which make no sense for paths.

I personally don't like the combination of "prominent default answer results in fixme" and "average user doesn’t know much about what his answer actually does" and "many of those fixme will only be able to be fixed on the ground" and "many of these continuing path/roads are not mapped for some reason, for example private property". This could easily lead to a huge mass of fixme that will never be worked on, but which prevents the few active local editor mappers from even looking at fixme tags.

Please provide an Overpass query for this proposal. Then we could probably drastically reduce the amount of useless tasks.


In urban areas we should completely skip
* highway=footway|path (1)

If the end node is within amenity=parking we should skip
* highway=service (2)

Of the remaining highways we should skip any way with
* man_made=pier (otherwise 100% noexit which is useless)

Of the remaining _end nodes_ we should skip any with
* highway=turning_circle (as mentioned above)
* highway=turning_loop (as mentioned above)
* noexit=yes (as mentioned above)
* fixme= (any value! Do not replace a "fixme=Here are steps leading south and upwards" with "fixme=continue"!)
* leisure=slipway (no need for noexit)
* amenity=parking_entrance (don't enforce micro mapping of underground parking)
* entrance= (any value, don't enforce indoor mapping)
* barrier= (for example barrier=gate seems to be common practice to state "here begins private property which is not mapped since there is no benefit of mapping it")


**Summary:**
Usecase A “pathes in the woods”: 
Please think again about guidelines “Effort vs impact”, “Easy answer”, “Publicly accessible” and “Valuable Surveyors” as these end nodes are rarely turning_loops, likely to fade out (impossible to map better and not noexit=yes), may continue but might already be tagged with a some fixme, can easily be spotted on maps already and most important: to fix this one has to visit the location on the ground again, and follow/track the path. Regarding "No spam" I'd guess that most of those ways would end up with a fixme= which is no improvement.

Usecase B “missing turning loops”:
Let’s assume we limit the quest to “highway=(living_street|pedestrian|residential)” and skip the end nodes mentioned above. Even then I’ve got a feeling that this quest might create far more problems than is solves. What to answer if there is a (missing) turning_loop, _and_ a missing path? And what should the average user answer when the path is more or less clearly private? What should he answer if there is no loop, but a private driveway? Please think again about guidelines “Users are no experts” and “Easy answer”.


**Footnotes:**
(1) skip highway=footway|path in urban areas, because of many cases like @HolgerJeromin mentioned in https://github.com/streetcomplete/StreetComplete/issues/3705#issuecomment-1028267285. Might otherwise demotivate mappers by enforcing a level of micromapping they did not want to do, while having hardly any advantages. noexit=yes obviously wrong, but fixme= will never get solved.
Similar cases mentioned by @andrewharvey in https://github.com/streetcomplete/StreetComplete/issues/3705#issuecomment-1026327588 with the list of areas to be excluded could be continued. Have a look at cemeteries for example.
Urban area in this context does include(!) smaller landuse=wood, gras and so on because people tend to map an area when some trees stand together...

(2) [highway=service](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dservice) without service= correctly tags the way accessing the parking area. In case there are no additional "highway=service + service=parking_aisle" mapped,(which is common for smaller parking areas) the end node has correctly no tag.

**Comment by mnalis** [comment]
That could probably be simplified as "_if the **ending node** of the way has **any tag at all**, skip this quest"_

As for the other suggestions, tags on the way (like `man_made=pier`) are probably cheap, but detecting if way is inside parking might be more expensive. Not to mention detecting if the way is "inside urban area" (we don't have that detection even in existing quests like `speed limit` quest, instead resorting to asking user explicitly)

**Comment by matkoniecz** [comment]
I looked at it again and I must say that I quite like this idea.

I will probably implement it either way for myself at least (though I would likely not add `fixme` tags, just switch to Vespucci)

> I am not exactly worrying that it would strictly invalid. I rather worry that it would be pointless/unwanted/disliked/irritating/unwelcome.

I am still worrying about how it would be liked/disliked by wider community. Especially as it would be the first quest adding `fixme` tags. What you think about asking on some wider forum such as https://community.openstreetmap.org/ and/or mailing list and maybe also proprietary discussion locations such as Discord/Slack?

(I could ask, but of someone else would ask what people think about SC adding `fixme` tags on a large scale)

**Comment by mpikzink** [comment]
Sure, feel free to start. I realize with the discussions that the quest will go over my head as a newbie anyway and I'm happy about your expertise and help.

**Comment by westnordost** [comment]
@matkoniecz could you outline what is your plan on implementing it? Can you summarize how you will tackle the issues mentioned in this thread? (P.S: I don't like the notion of StreetComplete adding fixme somewhere. Would your plan still work without it?)

**Comment by matkoniecz** [comment]
> Would your plan still work without it?

no

it fundamentally relies on it as one of options and is my primary reason why I am unsure is it a good idea

> "this is not the end of the road, it continues..."
> No : [fixme=continue](https://wiki.openstreetmap.org/wiki/DE:Key:fixme)

------------------

>  I don't like the notion of StreetComplete adding fixme somewhere

I guess that it can be WONTFIXed given this (if that is not actually instantly failing this quest - please reopen)

**Comment by westnordost** [comment]
I don't like it but if the broader community has no issue with it, then this is okay.

**Comment by westnordost** [comment]
@matkoniecz could you outline what is your plan on implementing it? Can you summarize how you will tackle the issues mentioned in this thread?

**Comment by matkoniecz** [comment]
Primary tricky cases:

This could result in SC mappers adding thousands, tens of thousands `fixme=continue`

Some people at least will grumble, maybe many of them.

Solution: ask people first and either WONTFIX this or in case of complaining after release: link discussions

(there is still risk of substantial complaints only after release or long-term complaints)

-----------------------------

For roads:

primary problem is cases where road ends and continues as driveways. Basically, case like here where driveway is not yet mapped

![screen02](https://user-images.githubusercontent.com/899988/162593012-1fe2e972-1746-4a1b-add4-1907de9ebe86.png)

in such case spamming `fixme=continue` would be appreciated only in places already deeply micromapped.

Solution: skip `highway=residential` `highway=service`

-----------------------------

For footways, where my plan is to basically "lets not implement it for footways":

> Thinking about edge cases like when a pathway leads into a park, it probably would trigger this but neither tag would make sense. You'd want to exclude paths that end inside areas like amenity=park, natural=beach

a) if `leisure=park ` area is very big SC could miss it (all elements for download outside bounding box used for download)
b) it may kill performance to run such checks

That is a primary issue where I have no convincing answer right now, I likely would look into a better tagging for such case.

> I think we will find many unconnected footways ending before a house like
> https://www.openstreetmap.org/way/220513514
> ![image](https://user-images.githubusercontent.com/2410353/152220166-bbe6a59b-85e7-44b6-82f5-597b54a241e8.png)

in such case path terminates nearby, but correct answer would be `fixme=continue`

Part of potential problem is that StreetComplete would be basically adding `fixme=please micromap this area`

---------------------

unknowns: as mentioned this may run into problems that were not initially predicted.

**Comment by westnordost** [comment]
> no residentials

I think this does not solve the issue you try to avoid and also takes away most of the usefulness of this quest:

Where to expect turning circles in non-residential areas? -> Commercial/Industrial districts, where the roads are often tagged as `highway=unclassified`. There, you'll also have the situation that at the end of the road, the road continues into the industrial landuse. I think, any place that potentially has a turning circle also has potentially abutters.

> no footways

Ok.  Wouldn't make sense otherwise

> fixme:continue

I agree that if you bring it up on the appropriate channels and they are OK with it, then `fixme:continue` is no problem.

It is truly a shame that `noexit=yes` means there is no exit whatsoever, for **any mode of transportation**. If there was a tag to mean "the road ends here", this could be set in cases where a footway or driveway "continues".

**Comment by westnordost** [comment]
Anyway, I will close this. It seems the subject is far more complicated/problematic to handle than initially assumed. And TBH, whether there is a turning circle somewhere can also usually be seen from aerial imagery.
